corehq/apps/app_manager/models.py

```diff
@@ -479,31 +557,54 @@ class OpenReferralAction(UpdateReferralAction):
 
 class OpenCaseAction(FormAction):
 
-    # `name_update` is the "official" version, while `name_update_multi` is intended as a temporary option
+    # `name_update` is the "official" version, while `name_update_multi` is intended as a way
     # to allow the user to resolve conflicts. They should not be used together. Either the action is in a
     # buildable state, where `name_update` is specified, or conflicts are waiting to be resolved, where
     # `name_updatd_multi` will hold the updates.
     name_update = SchemaProperty(ConditionalCaseUpdate)
     name_update_multi = SchemaListProperty(ConditionalCaseUpdate)
     external_id = StringProperty()
 
-    DIFF_VALUE_UPDATED = 'updated'
+    _NAME_UPDATE_MULTI_FIELD_NAME = 'name_update_multi'
 
-    def apply_updates(self, updates, diffs):
+    def apply_updates(self, updates, diffs, allow_conflicts=True):
         self.update_object(updates)
-        if self.DIFF_VALUE_UPDATED in diffs:
-            self.name_update = ConditionalCaseUpdate(diffs[self.DIFF_VALUE_UPDATED])
+        self.make_multi()
+
+        mappings = {self._NAME_UPDATE_MULTI_FIELD_NAME: self.name_update_multi}
+        normalized_diffs = diffs.convert_to_update_diff()
+        apply_diffs(mappings, normalized_diffs)
+        self.name_update_multi = mappings[self._NAME_UPDATE_MULTI_FIELD_NAME]
```

Comment:
Possibly could be cleaned up

Reply:
6319589f8708360d8b607d12c63a6b8b094ce3bd